bigbluebutton-html5/imports/ui/components/video-provider/hooks/index.ts

```diff
@@ -60,9 +62,29 @@ const useVideoStreamsSubscription = createUseSubscription(
 
 export const useStreams = () => {
   const { data, loading, errors } = useVideoStreamsSubscription();
-  const streams = useRef<Stream[]>([]);
+  const [streams, setStreams] = useState<Stream[]>([]);
 
-  if (loading) return streams.current;
+  useEffect(() => {
```

Comment:
I don't see why a `useEffect` along with a `useState` is necessary here. This hook should be pure. The `ref` was storing the last state to be returned when the subscription was `loading`.

Reply:
what do you mean with `pure`? I decoupled the logic from the main function for better readability and follow the react patterns.

Reply:
Sorry, by "pure" I mean no effects, no state. It should only map the return value of `useVideoStreamsSubscription()` and return it (or return the last mapped value when `loading` is `true`). I think placing this logic inside a `useEffect` and then updating the state will make this hook render twice.